python/pyspark/pandas/tests/test_series.py

```diff
@@ -2209,12 +2209,12 @@ def test_mad(self):
         pser = pd.Series([1, 2, 3, 4], name="Koalas")
         psser = ps.from_pandas(pser)
 
-        self.assert_eq(pser.mad(), psser.mad())
+        self.assert_eq(pser.mad(), psser.mad(), almost=True)
```

Comment:
Could you try with `check_exact=False` instead of `almost=True`?
If it works, it's better because it still checks the internal structure of the Series / DataFrame; otherwise `almost=True` checks only values.

Reply:
Yep, I tried `self.assert_eq(pser.mad(), psser.mad(), check_exact=False)`, it failed again. Because the test.mad() result value type is a numpy.float64 scalar, it would be matched to:

https://github.com/apache/spark/blob/c38c617bcd04429e50310e4c7e0bd989c6555836/python/pyspark/testing/pandasutils.py#L237-L240

So, only `almost=True` work.

And I know your concern it's make sure the type is right in here, but not only to check value. I notice that only last check is failed in my env, so we can keep top 3 check as original, and just make last check as almost=True.

Reply:
And one more thing, this test case failed in my env always, even the latest pandas v1.3.4, so looks like a precision issue cross platfroms.

Reply:
Addressed.

Reply:
Ah, got it. The result was not `DataFrame` or `Series`, then `almost=True` should be fine. Thanks.